admin/c2cgeoportal_admin/schemas/metadata.py

```diff
@@ -0,0 +1,102 @@
+import colander
+from deform.widget import MappingWidget, SelectWidget, SequenceWidget, TextAreaWidget
+from c2cgeoform.schema import GeoFormSchemaNode
+from c2cgeoportal_commons.models.main import Metadata
+
+
+@colander.deferred
+def metadata_types(node, kw):
+    return {
+        m['name']: m.get('type', 'string')
+        for m in kw['request'].registry.settings['admin_interface']['available_metadata']
+    }
+
+
+@colander.deferred
+def metadata_name_widget(node, kw):
+    return SelectWidget(
+        values=[
+            (m['name'], m['name'])
+            for m in sorted(
+                kw['request'].registry.settings['admin_interface']['available_metadata'],
+                key=lambda m: m['name'])
+        ]
+    )
+
+
+class MetadataSchemaNode(GeoFormSchemaNode):
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+
+        self.available_types = []
+
+        self._add_value_node('string', colander.String())
+        self._add_value_node('liste', colander.String())
+        self._add_value_node('boolean', colander.Boolean())
+        self._add_value_node('int', colander.Int())
+        self._add_value_node('float', colander.Float())
+        self._add_value_node('url', colander.String())
+        self._add_value_node(
+            'json',
+            colander.String(),
+            widget=TextAreaWidget(rows=10))
+        # self._add_value_node('date', colander.Date())
+        # self._add_value_node('time', colander.Time())
+        # self._add_value_node('datetime', colander.DateTime())
+
+    def _add_value_node(self, type_name, colander_type, **kw):
+        self.add_before(
+            'description',
+            colander.SchemaNode(
+                colander_type,
+                name=type_name,
+                missing=colander.null,
+                **kw))
+        self.available_types.append(type_name)
+
+    def objectify(self, dict_, context=None):
+        # depending on the type get the value from the right widget
+        dict_['value'] = dict_[self._ui_type(dict_['name'])]
+        return super().objectify(dict_, context)
+
+    def dictify(self, obj):
+        dict_ = super().dictify(obj)
+        value = obj.value or colander.null
+        '''
+        import dateutil.parser
+        ui_type = self._ui_type(obj.name)
+        if ui_type == 'date':
+            value = dateutil.parser.parse(value).date()
+        if ui_type == 'datetime':
+            value = dateutil.parser.parse(value)
+        if ui_type == 'time':
+            value = dateutil.parser.parse(value).time()
+        '''
```

Comment:
A dict could be used ? dateutil.parser.parse(value) isolated ? an utility class used ?

Reply:
maybe, when we will uncomment these lines, not really useful for the moment.